### PR TITLE
assigns autocompleter value to angular scope

### PR DESCRIPTION
**Pull request by ulferts**
https://www.openproject.org/work_packages/15643

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-relations-autocompleter-value-assignment.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

**Comment by myabc** [comment]
@ulferts I took another look at this, with the aim of getting a cleaner solution. My preferred method would be to add a `change` event listener on the input element from within the `addWorkPackageRelation`. I haven't managed to get this to work yet though…

File: app/assets/javascripts/angular/work_packages/view_models/relations-handler.js [hunk 1]
```diff
@@ -98,7 +98,12 @@ angular.module('openproject.viewModels')
                                    frequency: 0.5,
                                    paramName: 'q',
                                    updateElement: function(value) {
+                                     // Have to use the duplicate assignment here to update the field
+                                     // * to the user
+                                     // * to the angular scope
+                                     // Doing just one will not suffice.
                                      document.getElementById('relation_to_id-' + relationsId).value = value.id;
+                                     jQuery('#relation_to_id-' + relationsId).scope().relationToAddId = value.id;
```
Comment (myabc) [inline app/assets/javascripts/angular/work_packages/view_models/relations-handler.js]: We shouldn't really mix `document.getElementById`, `angular.element` and `jQuery` APIs. I think it's probably best to standardise on `angular.element` (you'd also have the added benefit of being able to chain here)

**Comment by ulferts** [comment]
@myabc, I altered the solution according to your suggestion. Thanks for the tip.

While fixing, I stumbled over another bug for which 09b1d0c is a working but dirty fix. When the relations tab is loaded right away, e.g. this url https://openproject-org.qa.finn.de/projects/openproject/work_packages/19901/relations, the directive and the applyCustomExtensions function gets executed before the necessary DOM exists. Any tips on that?

**Pull request merged**